hive-metastore/src/main/java/org/apache/iceberg/hive/HiveViewOperations.java

```diff
@@ -0,0 +1,386 @@
+        switch (commitStatus) {
+          case SUCCESS:
+            break;
+          case FAILURE:
+            throw e;
+          case UNKNOWN:
+            throw new CommitStateUnknownException(e);
+        }
+      }
+    } catch (TException e) {
+      throw new RuntimeException(
+          String.format("Metastore operation failed for %s.%s", database, viewName), e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during commit", e);
+
+    } catch (LockException e) {
+      throw new CommitFailedException(e);
+
+    } finally {
+      HiveOperationsBase.cleanupMetadataAndUnlock(io(), commitStatus, newMetadataLocation, lock);
+    }
+
+    LOG.info(
+        "Committed to view {} with the new metadata location {}", fullName, newMetadataLocation);
+  }
+
+  /**
+   * Validate if the new metadata location is the current metadata location.
+   *
+   * @param newMetadataLocation newly written metadata location
+   * @return true if the new metadata location is the current metadata location
+   */
+  private boolean checkCurrentMetadataLocation(String newMetadataLocation) {
+    ViewMetadata metadata = refresh();
+    return newMetadataLocation.equals(metadata.metadataFileLocation());
+  }
+
+  private void setHmsTableParameters(
+      String newMetadataLocation, Table tbl, ViewMetadata metadata, Set<String> obsoleteProps) {
+    Map<String, String> parameters =
+        Optional.ofNullable(tbl.getParameters()).orElseGet(Maps::newHashMap);
+
+    // push all Iceberg view properties into HMS
+    metadata.properties().entrySet().stream()
+        .filter(entry -> !entry.getKey().equalsIgnoreCase(HiveCatalog.HMS_TABLE_OWNER))
+        .forEach(entry -> parameters.put(entry.getKey(), entry.getValue()));
+    if (metadata.uuid() != null) {
+      parameters.put("uuid", metadata.uuid());
+    }
+
+    // remove any props from HMS that are no longer present in Iceberg view props
+    obsoleteProps.forEach(parameters::remove);
+
+    parameters.put(
+        BaseMetastoreTableOperations.TABLE_TYPE_PROP,
+        ICEBERG_VIEW_TYPE_VALUE.toUpperCase(Locale.ENGLISH));
+    parameters.put(BaseMetastoreTableOperations.METADATA_LOCATION_PROP, newMetadataLocation);
+
+    if (currentMetadataLocation() != null && !currentMetadataLocation().isEmpty()) {
+      parameters.put(
+          BaseMetastoreTableOperations.PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation());
+    }
+
+    // If needed set the 'storage_handler' property to enable query from Hive
+    parameters.remove(hive_metastoreConstants.META_TABLE_STORAGE);
+
+    setSchema(metadata.schema(), parameters);
+    tbl.setParameters(parameters);
+  }
+
+  private static boolean hiveLockEnabled(Configuration conf) {
+    return conf.getBoolean(ConfigProperties.LOCK_HIVE_ENABLED, true);
+  }
+
+  private Table newHMSView(ViewMetadata metadata) {
```

Comment:
@nastra , To use [newHMSTable](https://github.com/apache/iceberg/blob/main/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java#L215). We should have options to support extra parameters there. What do you recommend ?